libs/forcetk.mobilesdk.js

```diff
@@ -299,56 +316,61 @@ if (forcetk.Client === undefined) {
      */
     var ajaxRequestId = 0;
     forcetk.Client.prototype.ajax = function(path, callback, error, method, payload, headerParams) {
-        var tag = "";
-        var that = this;
-        var retryCount = 0;
-        var url = (path.indexOf(this.instanceUrl) == 0 ? path : this.instanceUrl + (path.indexOf('/services/data') == 0 ? path : '/services/data' + path));
-        return $j.ajax({
-            type: method || "GET",
-            async: this.asyncAjax,
-            url: (this.proxyUrl !== null) ? this.proxyUrl: url,
-            contentType: method == "DELETE" || method == "GET" ? null : 'application/json',
-            cache: false,
-            processData: false,
-            dataType: "json",
-            data: payload,
-            headers: getRequestHeaders(this),
-            success: function() {
-                console.timeEnd(tag);
-                callback.apply(null, arguments);
-            },
-            error: function(jqXHR, textStatus, errorThrown) {
-                console.timeEnd(tag);
-                var xhr = this;
-                var errorCallback = function() {
-                    if (typeof error == 'function') {
-                        error(jqXHR, textStatus, errorThrown);
+        if (this.useNativeNetworking) {
+            cordova.require("com.salesforce.plugin.network").sendRequest('/services/data', path, callback, error, method, payload, headerParams);
```

Comment:
@wmathurin This is what I'm proposing. A simple `if-else`. The `else` block is the old implementation with no changes.